codegen/gateway.go

```diff
@@ -462,6 +468,8 @@ type EndpointSpec struct {
 	ClientMethod string
 	// The client for this endpoint if httpClient or tchannelClient
 	ClientSpec *ClientSpec
+	// check if should log the request of service to client
+	LogDownstreamRequest LogDownstreamRequestSpec
```

Comment:
i agree that this approach is not the most ideal and does need build and deploy. Since the change will mostly be one time (a few times) for debugging shadow result, and needs to be unblocked pretty soon. Besides the ideal solution you proposed, is there a better way we can keep a balance between urgency and work load?